File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long s = 0, e = (long) std::sqrt(c);
        while (s <= e) {
            long sum = s * s + e * e;
            if (sum < c) {
                s++;
            } else if (sum > c) {
                e--;
            } else {
                return true;
            }
        }
        return c == 1;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        s, e = 0, int(c**0.5)
        while s <= e:
            sum = s * s + e * e
            if sum < c:
                s += 1
            elif sum > c:
                e -= 1
            else:
                return True
        return c == 1